eliot/_parse.py

```diff
@@ -38,7 +45,22 @@ def _insert_action(self, node):
 
         @return: Updated L{Task}.
         """
-        task = self.transform(["_nodes", node.task_level], node)
+        task = self
+        if (node.end_message and node.start_message
+            and (len(node.children) ==
+                 node.end_message.task_level.level[-1] - 2)):
+            # Possibly this action is complete, make sure all sub-actions
+            # are complete:
+            completed = True
+            for child in node.children:
+                if (isinstance(child, WrittenAction) and
```

Comment:
:grey_question: Usual concern about using isinstance instead of OO.  Looks like it's only once in this file, so probably OK as is. Otherwise making both WrittenActions and WrittenMessages implement an interface that has a "fragmentable" attribute or something that's true for Actions (things that are split up across lines) and false for Messages (things that are always contiguous on a single line). might be a quick way out of this. Probably too much overhead for now though.